lib/mongo/protocol/message.rb

```diff
@@ -226,7 +226,7 @@ def serialize_fields(buffer, max_bson_size = nil)
       # @param buffer [String] Buffer to receive the header
       # @return [String] Serialized header
       def serialize_header(buffer)
-        set_request_id unless @request_id
+        set_request_id unless defined?(@request_id)
```

Comment:
ok. How about we set the ``@request_id`` to nil in the initialize method and use the memoize code I suggested for #set_request_id?